JDBC Source created representation for numeric cannot be consumed by MongoDB Sink

### What version of the Stream Reactor are you reporting this issue for?
Stream Reactor 1.2.4  (I'm using the lensesio/box docker image)

### Are you running the correct version of Kafka/Confluent for the Stream reactor release?
I hope that lensesio/box use the correct version.

### Do you have a supported version of the data source/sink .i.e Cassandra 3.0.9?
Running mongo:4.2-bionic container.

### What is the expected behaviour?
I have a postgres database and use the JDBC Source connector to create a Kafka topic from it. Then I want to use the MongoDB Sink connector to pipe the data into my mongodb.

In the postgres database, I have a numeric(8,2) field. This is translated into the following avro schema in kafka by the jdbc source connector:

```
{
  "name": "price",
  "type": [
	"null",
	{
	  "type": "bytes",
	  "scale": 2,
	  "precision": 64,
	  "connect.version": 1,
	  "connect.parameters": {
		"scale": "2"
	  },
	  "connect.name": "org.apache.kafka.connect.data.Decimal",
	  "logicalType": "decimal"
	}
  ],
  "default": null
}
```

I would expect this to work with the MongoDB sink and be represented as a a [decimal bson type](https://docs.mongodb.com/manual/tutorial/model-monetary-data/#numeric-decimal) existing in MongoDB since version 3.4.

### What was observed?
I get the following exception:

```
org.apache.kafka.connect.errors.ConnectException: Exiting WorkerSinkTask due to unrecoverable exception.
	at org.apache.kafka.connect.runtime.WorkerSinkTask.deliverMessages(WorkerSinkTask.java:561)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.poll(WorkerSinkTask.java:322)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.iteration(WorkerSinkTask.java:224)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.execute(WorkerSinkTask.java:192)
	at org.apache.kafka.connect.runtime.WorkerTask.doRun(WorkerTask.java:175)
	at org.apache.kafka.connect.runtime.WorkerTask.run(WorkerTask.java:219)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.RuntimeException: org.apache.kafka.connect.errors.DataException: Invalid type for BYTES: class java.math.BigDecimal
	at com.datamountaineer.streamreactor.connect.errors.ThrowErrorPolicy.handle(ErrorPolicy.scala:58)
	at com.datamountaineer.streamreactor.connect.errors.ErrorHandler$class.handleError(ErrorHandler.scala:83)
	at com.datamountaineer.streamreactor.connect.errors.ErrorHandler$class.handleTry(ErrorHandler.scala:64)
	at com.datamountaineer.streamreactor.connect.mongodb.sink.MongoWriter.handleTry(MongoWriter.scala:40)
	at com.datamountaineer.streamreactor.connect.mongodb.sink.MongoWriter.insert(MongoWriter.scala:135)
	at com.datamountaineer.streamreactor.connect.mongodb.sink.MongoWriter.write(MongoWriter.scala:66)
	at com.datamountaineer.streamreactor.connect.mongodb.sink.MongoSinkTask$$anonfun$put$2.apply(MongoSinkTask.scala:72)
	at com.datamountaineer.streamreactor.connect.mongodb.sink.MongoSinkTask$$anonfun$put$2.apply(MongoSinkTask.scala:72)
	at scala.Option.foreach(Option.scala:257)
	at com.datamountaineer.streamreactor.connect.mongodb.sink.MongoSinkTask.put(MongoSinkTask.scala:72)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.deliverMessages(WorkerSinkTask.java:539)
	... 10 more
Caused by: org.apache.kafka.connect.errors.DataException: Invalid type for BYTES: class java.math.BigDecimal
	at com.datamountaineer.streamreactor.connect.mongodb.converters.SinkRecordConverter$.com$datamountaineer$streamreactor$connect$mongodb$converters$SinkRecordConverter$$convertToDocument$1(SinkRecordConverter.scala:225)
	at com.datamountaineer.streamreactor.connect.mongodb.converters.SinkRecordConverter$$anonfun$com$datamountaineer$streamreactor$connect$mongodb$converters$SinkRecordConverter$$convertToDocument$1$3.apply(SinkRecordConverter.scala:187)
	at com.datamountaineer.streamreactor.connect.mongodb.converters.SinkRecordConverter$$anonfun$com$datamountaineer$streamreactor$connect$mongodb$converters$SinkRecordConverter$$convertToDocument$1$3.apply(SinkRecordConverter.scala:186)
	at scala.collection.TraversableOnce$$anonfun$foldLeft$1.apply(TraversableOnce.scala:157)
	at scala.collection.TraversableOnce$$anonfun$foldLeft$1.apply(TraversableOnce.scala:157)
	at scala.collection.Iterator$class.foreach(Iterator.scala:891)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1334)
	at scala.collection.IterableLike$class.foreach(IterableLike.scala:72)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
	at scala.collection.TraversableOnce$class.foldLeft(TraversableOnce.scala:157)
	at scala.collection.AbstractTraversable.foldLeft(Traversable.scala:104)
	at com.datamountaineer.streamreactor.connect.mongodb.converters.SinkRecordConverter$.com$datamountaineer$streamreactor$connect$mongodb$converters$SinkRecordConverter$$convertToDocument$1(SinkRecordConverter.scala:186)
	at com.datamountaineer.streamreactor.connect.mongodb.converters.SinkRecordConverter$.fromStruct(SinkRecordConverter.scala:234)
	at com.datamountaineer.streamreactor.connect.mongodb.sink.SinkRecordToDocument$.apply(SinkRecordToDocument.scala:66)
	at com.datamountaineer.streamreactor.connect.mongodb.sink.MongoWriter$$anonfun$insert$2$$anonfun$apply$2.apply(MongoWriter.scala:86)
	at com.datamountaineer.streamreactor.connect.mongodb.sink.MongoWriter$$anonfun$insert$2$$anonfun$apply$2.apply(MongoWriter.scala:82)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
	at scala.collection.Iterator$class.foreach(Iterator.scala:891)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1334)
	at scala.collection.IterableLike$class.foreach(IterableLike.scala:72)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
	at scala.collection.TraversableLike$class.map(TraversableLike.scala:234)
	at scala.collection.AbstractTraversable.map(Traversable.scala:104)
	at com.datamountaineer.streamreactor.connect.mongodb.sink.MongoWriter$$anonfun$insert$2.apply(MongoWriter.scala:82)
	at com.datamountaineer.streamreactor.connect.mongodb.sink.MongoWriter$$anonfun$insert$2.apply(MongoWriter.scala:78)
	at scala.collection.immutable.Map$Map1.foreach(Map.scala:116)
	at com.datamountaineer.streamreactor.connect.mongodb.sink.MongoWriter.insert(MongoWriter.scala:78)
	... 16 more
```

### What is your Connect cluster configuration (connect-avro-distributed.properties)?
I just use this compose config (note, you have to add your lensesio key):
```
 kafka-lenses:
    image: lensesio/box
    restart: on-failure
    ports:
      - 3030:3030
      - 9092:9092
      - 8081:8081
      - 8082:8082
      - 8083:8083
    environment:
      SAMPLEDATA: 0
      ADV_HOST: 127.0.0.1
      EULA: "https://dl.lenses.io/d/?id=<your-lensesio-key>"
```

--- a/kafka-connect-mongodb/src/main/scala/com/datamountaineer/streamreactor/connect/mongodb/converters/SinkRecordConverter.scala
+++ b/kafka-connect-mongodb/src/main/scala/com/datamountaineer/streamreactor/connect/mongodb/converters/SinkRecordConverter.scala
@@ -122,6 +122,7 @@ object SinkRecordConverter extends StrictLogging {
               case Schema.Type.BYTES =>
                 if (schema != null && Decimal.LOGICAL_NAME == schema.name) {
                   value match {
+                    case jbd: java.math.BigDecimal => jbd
                     case bd: BigDecimal => bd.bigDecimal
                     case bb: ByteBuffer => Decimal.toLogical(schema, bb.array())
                     case _ => Decimal.toLogical(schema, value.asInstanceOf[Array[Byte]])

